packages/analytics/amplify_analytics_pinpoint_dart/test/event_client_test.dart

```diff
@@ -0,0 +1,406 @@
+          () => PutEventsResponse(
+            eventsResponse: EventsResponse(
+              results: {
+                mockEndpointId: ItemResponse(
+                  endpointItemResponse: EndpointItemResponse(
+                    message: 'message',
+                    statusCode: 200,
+                  ),
+                  eventsItemResponse: {
+                    '0': EventItemResponse(
+                      message: 'Accepted',
+                      statusCode: 200,
+                    ),
+                    // TODO(kylechen): retryable exceptions should only be status code >=500 <600
+                    '1': EventItemResponse(
+                      message: 'Retryable Exception',
+                      statusCode: 500,
+                    )
+                  },
+                ),
+              },
+            ),
+          ),
+        ),
+      );
+
+      await eventClient.flushEvents();
+
+      final items = eventStore.getCount(100);
+      expect(items.length, 1);
+
+      final storedEvent = StoredEvent(items.elementAt(0), jsonSerializers);
+      final event = storedEvent.event;
+
+      expect(event.eventType, retryableFailEventType);
+    });
+
+    test('flushEvents deletes events with non retryable exception', () async {
+      await eventClient.recordEvent(
+        eventType: successEventType,
+      );
+      await eventClient.recordEvent(
+        eventType: failEventType,
+      );
+
+      when(() => pinpointClient.putEvents(any<PutEventsRequest>())).thenReturn(
+        mockSmithyOperation(
+          () => PutEventsResponse(
+            eventsResponse: EventsResponse(
+              results: {
+                mockEndpointId: ItemResponse(
+                  endpointItemResponse: EndpointItemResponse(
+                    message: 'message',
+                    statusCode: 200,
+                  ),
+                  eventsItemResponse: {
+                    '0': EventItemResponse(
+                      message: 'Accepted',
+                      statusCode: 200,
+                    ),
+                    // TODO(kylechen): retryable exceptions should only be status code >=500 <600
+                    '1': EventItemResponse(
+                      message: 'ValidationException',
+                      statusCode: 400,
+                    )
+                  },
+                ),
+              },
+            ),
+          ),
+        ),
+      );
+
+      await eventClient.flushEvents();
+
+      final items = eventStore.getCount(100);
+      expect(items.length, 0);
+    });
+
+    test(
+        'flushEvents does not delete events if pinpointClient throws retryable exception',
+        () async {
+      await eventClient.recordEvent(
+        eventType: successEventType,
+      );
+      await eventClient.recordEvent(
+        eventType: failEventType,
+      );
+
+      when(() => pinpointClient.putEvents(any<PutEventsRequest>()))
+          .thenThrow(BadRequestException());
```

Comment:
Should be the smithy operation completes with this error - `putEvents` never directly throws

Reply:
I'm saying `putEvents` doesn't throw. `putEvents` returns a SmithyOperation which when you await its `result` property throws.

Reply:
Okay understood, updated the tests.